coding-the-legislation/20_input_variables.md

```diff
@@ -31,4 +31,62 @@ class french_citizen(Variable):
     definition_period = YEAR
 ```
 
-The default default value of a `Variable` is `0` for numeric variables and  `False` for boolean ones.
+If you do not explicitly define a default value, the following will be used:
+
+  - For numeric variables: `0`.
+  - For boolean variables: `False`.
+  - For enum variables: the item of index `0`.
+
+## Advanced example
```

Comment:
Proposition : 

## Advanced example:  enumerations (enum)

### When to use enums ?
Let's say you want to calculate a housing tax that depend on the occupancy status of the inhabitants.  
[INSERT FORMULA]
The `housing_occupancy_status` will be an input variable, and have a limited set of possible inputs (e.g. Owner, Tenant ...). We can express this through the enum type.

### how to use the enum in an input variable
1. create an enum `HOUSING_OCCUPANCY_STATUS`
```py
 HOUSING_OCCUPANCY_STATUS = Enum([
     u'Tenant',
     u'Owner',
     u'Free lodger',
     u'Homeless'])
 ```
Enum items can be references by their index (stating at `0`), or their string definition
`HOUSING_OCCUPANCY_STATUS['Owner']` is equivalent to `HOUSING_OCCUPANCY_STATUS['1'] ` ** [A vérifier 🤔 :]**

2. Create a variable `housing_occupancy_status`

```py
 class housing_occupancy_status(Variable):
     column = EnumCol(
         enum = HOUSING_OCCUPANCY_STATUS
         default = 1  # 'Owner'
         )
     entity = Household
     definition_period = MONTH
     label = u"Legal housing situation of the household concerning their main residence"
 ```

3. Use the enum in a variable formula

To get `housing_occupancy_status` for a given `month` , call `household('housing_occupancy_status', month)`.
[Insert formula here]

4. Test the formula by inputing the enum
A YAML test would look as follows:

 ```yaml
  name: Household with free lodger status living in a 100 sq.meters accomodation
   period: 2017
   input_variables:
     accomodation_size:
       2017-01: 100
     housing_occupancy_status:
       2017-01: Free lodger
   output_variables:
     housing_tax: 0

```
In your tests and calls to the Web API, favor the enum's string definition as it is more readable than the enumeration index.